src/main/scala/fi/oph/koski/valpas/sso/ValpasOppijaCasServlet.scala

```diff
@@ -0,0 +1,55 @@
+package fi.oph.koski.valpas.sso
+
+import fi.oph.koski.config.KoskiApplication
+import fi.oph.koski.http.KoskiErrorCategory
+import fi.oph.koski.koskiuser.{AuthenticationUser, UserLanguage}
+import fi.oph.koski.log.LogUserContext
+import fi.oph.koski.servlet.NoCache
+import fi.oph.koski.sso.CasOppijaCreationService
+import fi.oph.koski.valpas.servlet.{ValpasApiServlet, ValpasBaseServlet}
+import fi.oph.koski.valpas.valpasuser.ValpasAuthenticationSupport
+
+class ValpasOppijaCasServlet(implicit val application: KoskiApplication) extends ValpasApiServlet with ValpasBaseServlet with ValpasAuthenticationSupport with NoCache {
+  private val koskiSessions = application.koskiSessionRepository
+  private val casService = application.casService
+  private val oppijaCreation = new CasOppijaCreationService(application)
+
+  def virhesivu: String = valpasRoot + "/virhe/login"
+  def eiTietojaOpintopolussaSivu: String = valpasRoot + "/eitietoja"
+
+  get("/oppija") {
+    params.get("ticket") match {
+      case Some(ticket) =>
+        try {
+          val hetu = casService.validateKansalainenServiceTicket(casValpasOppijaServiceUrl, ticket)
+          oppijaCreation.findOrCreate(request, hetu) match {
+            case Some(oppija) =>
+              val huollettavat = application.huoltajaServiceVtj.getHuollettavat(hetu)
```

Comment:
tämä huoltajaServiceVtj:kin voisi olla luokan privaattina kenttänä. On vähän hämäävää koodinlukijalle, kun osa on erillisinä kenttinä, ja osaa käytetään application:in kautta.

Reply:
Sama juttu alempana application.directoryClient:in kanssa.